orttraining/orttraining/python/training/_ortmodule_io.py

```diff
@@ -5,6 +5,55 @@
 import torch
 import warnings
 
+
+class _InputInfo(object):
+    def __init__(self, names, shape, require_grad_names = [], dynamic_axes = {}, schema = []):
+        self.names = names
+        self.shape = shape
+        self.require_grad_names = require_grad_names
+        self.dynamic_axes = dynamic_axes
+        self.schema = schema
```

Comment:
I'm surprised our linter doesn't complain about this, but it's generally a good idea to use `None` as the defaults for potentially-mutable default args, just in case someone comes along and modifies the default and creates a hard-to-debug issue:

```suggestion
    def __init__(self, names, shape, require_grad_names=None, dynamic_axes=None, schema=None):
        self.names = names
        self.shape = shape
        self.require_grad_names = require_grad_names if require_grad_names else []
        self.dynamic_axes = dynamic_axes if dynamic_axes else {}
        self.schema = schema if schema else []
```

Alternatively, since there's only one callsite, and it passes all of the arguments, you could make all of them mandatory.